src/parser.ts

```diff
@@ -187,9 +183,197 @@ function getFunctionName(func: ts.FunctionLikeDeclaration): string {
+    if (!member) {
+        return null;
+    }
+    // if the function is a property, and either has a local name,
+    // or is inside a class expression, we take the local name and
+    // and do not prefix it with the object literal chain
+    if (isProperty) {
+        if (func.name !== undefined || ts.isClassExpression(member.parent)) {
+            return null;
+        }
+    }
+    return getObjectLiteralPrefixR(member.parent);
+}
+
+function getObjectLiteralPrefixR(
+    node: ts.Node,
+    prefixWithObject = false
+): string | null {
+    const parent = node.parent;
+    if (!parent) {
+        return null;
+    }
+    if (ts.isPropertyAssignment(parent)) {
+        const propertyAssignment = parent;
+        const prefix = getObjectLiteralPrefixR(
+            propertyAssignment.parent,
+            true
+        );
+        return `${prefix}.${getPropertyName(propertyAssignment.name)}`;
+    } else if (ts.isVariableDeclaration(parent)) {
+        if (ts.isIdentifier(parent.name)) {
+            return getPropertyName(parent.name);
+        }
+    } else if (
+        ts.isBinaryExpression(parent) &&
+        parent.operatorToken.kind === ts.SyntaxKind.EqualsToken
+    ) {
+        if (ts.isIdentifier(parent.left)) {
+            return getPropertyName(parent.left);
+        } else if (
+            ts.isPropertyAccessExpression(parent.left) ||
+            ts.isElementAccessExpression(parent.left)
+        ) {
+            return getLeftHandSideName(parent.left);
+        }
+    }
+    return prefixWithObject ? "<Object>" : null;
+}
+
+function getParentClassProperty(func: ts.FunctionLikeDeclaration) {
+    let { member } = getMemberNodeForFunction(func);
+    if (member) {
+        const parent = member.parent;
+        if (ts.isClassDeclaration(parent) || ts.isClassExpression(parent)) {
+            return parent;
+        }
+    }
+    return null;
+}
+
+function getClassPrefix(func: ts.FunctionLikeDeclaration): string | null {
+    let { member, isProperty } = getMemberNodeForFunction(func);
+    if (member) {
+        const parent = member.parent;
+        if (ts.isClassDeclaration(parent) || ts.isClassExpression(parent)) {
+            const className = getClassName(parent);
+            if (isStatic(member)) {
+                if (!isProperty || func.name === undefined) {
+                    return className;
+                }
+            } else if (ts.isConstructorDeclaration(func)) {
+                return className;
+            } else if (!isProperty) {
+                return `${className}.prototype`;
+            }
+        }
+    }
+    return null;
+}
+
+function getClassName(classNode: ts.ClassLikeDeclaration): string {
+    // if class has a local name, use it and return,
+    // not interested in left hand side
+    if (classNode.name) {
+        return classNode.name.text;
+    }
+    // try to get the name from the left hand side
+    let nameText = null;
```

Comment:
Please do not use an indirect return value. Instead, return directly from the two positive cases. Then the final line can unconditionally return the default.